typings/glamorous.d.ts

```diff
@@ -3,32 +3,58 @@
 // Definitions by: Kok Sam <https://github.com/sammkj>
 
 import * as React from 'react';
-import { ComponentClass, CSSProperties, StatelessComponent, SVGAttributes } from 'react'
+import {
+  HTMLGlamorousInterface,
+  SVGGlamorousInterface,
+} from "./element-interfaces.d";
 
-import { HTMLGlamorousInterface, SVGGlamorousInterface } from './element-interfaces'
+type GlamorousComponent<P> = React.ComponentClass<P> | React.StatelessComponent<P>;
 
-export type Component = ComponentClass<{}> | StatelessComponent<{}>;
+export type StaticStyles<Properties> = Partial<Properties>
+export type DynamicStyledFunction<Properties> = (
+    props?: object,
+    theme?: object
+  ) => Partial<Properties>;
 
-export interface DynamicStyledFunction {
-  (props?: object, theme?: object): object;
+export interface StyledFunction<Props, Properties> {
+  (style1: DynamicStyledFunction<Properties>, style2: StaticStyles<Properties>, ...styles: Array<DynamicStyledFunction<Properties> | StaticStyles<Properties>>): GlamorousComponent<Props>;
 }
 
-export interface StyledFunction<Properties> {
+// TODO: Using a union for a parameter kills autocomplete so we
+// use overloading to give autocomplete on the first two styles
+// and retain typesafety on additional styles
+export interface StyledFunction<Props, Properties> {
   (
-    staticStyles: Partial<Properties>,
-    dynamicStyles?: DynamicStyledFunction
-  ): Component;
+    style1: StaticStyles<Properties>,
+    style2?: DynamicStyledFunction<Properties>,
+    ...styles: Array<DynamicStyledFunction<Properties> | StaticStyles<Properties>>
+  ): GlamorousComponent<Props>;
 }
 
-export interface GlamorousBaseInterface {
-  (component: Component): StyledFunction<CSSProperties | SVGAttributes<any>>;
+export interface StyledFunction<Props, Properties> {
```

Comment:
Having multiple signatures in the same interface also loses the autocomplete.  Overloading in this way is the only way I'm aware of to keep autocomplete.